packages/elements/src/slider/elements/slider.ts

```diff
@@ -951,20 +951,29 @@ export class Slider extends FormFieldElement {
    */
   private updateNotifyProperty(name: SliderDataName, value: string): void {
     let shouldUpdate = false;
-
-    if (name === SliderDataName.to) {
-      shouldUpdate = this.isValueInBoundary(Number(value), SliderDataName.to);
-    } else {
-      shouldUpdate = this.isValueInBoundary(Number(value), '');
-    }
+    const valueNumber = Number(value);
+    shouldUpdate = this.isValueInBoundary(valueNumber, name);
+
+    // Adjust update-pending from/to value based on min range
+    const adjustValue = (value: string): string => {
+      const _currentValue = Number(value);
+      if (name === SliderDataName.from && _currentValue === this.toNumber) {
+        return String(_currentValue - this.minRangeNumber);
+      }
+      if (name === SliderDataName.to && _currentValue === this.fromNumber) {
+        return String(_currentValue + this.minRangeNumber);
+      }
+      return value;
+    };
 
     if (shouldUpdate) {
-      this[name] = value;
-      this.notifyPropertyChange(name, value);
-    } else {
-      const inputName = `${name}Input`;
-      this[inputName as NumberFieldName].value = this[name];
+      this[name] = adjustValue(this.format(valueNumber));
+      this.notifyPropertyChange(name, this[name]);
     }
+
+    // Revert number field value back to the pre-update value
```

Comment:
Actually, "revert" doesn't completely cover what is happening here. My bad. Let's revise the comment.
```suggestion
    // Sync number field value with its component property counterpart
```